pkg/splunk/client_test.go

```diff
@@ -190,3 +193,37 @@ func TestSplunkLoggerPayloads(t *testing.T) {
 		})
 	}
 }
+
+func TestSplunkLoggerCloseTimeout(t *testing.T) {
+	stopCh := make(chan struct{})
+
+	var handlerCalls atomic.Int32
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		handlerCalls.Add(1)
+		select {
+		case <-time.After(10 * time.Second):
+			return
+		case <-stopCh:
+			return
+		}
+	}))
+	defer srv.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel()
+
+	sl := newSplunkLogger(ctx, srv.URL, "token", "source", "hostname", 0)
+	_, err := sl.event([]byte("{}\n"))
+	if err != nil {
+		t.Error(err)
+	}
+	start := time.Now()
+	closeTimeout := 100 * time.Millisecond
+
+	err = sl.close(closeTimeout)
+	assert.Equal(t, ErrCloseTimeout, err)
+	assert.True(t, time.Since(start) < 5*closeTimeout)
```

Comment:
Having some comment here (like `// add a lot of slack to the closeTimeout to account for slow  GH VMs that may run the tests`  would be nice I think, just to make sure there is no wondering why we are not checking more exactly for the timeout.